### PR TITLE
Add balance control for the LP

### PR DESCRIPTION
**Pull request by aguschirico**

### CODE REVIEW

File: http/server.go [hunk 1]
```diff
@@ -216,18 +216,19 @@ func (s *Server) acceptQuoteHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	p := getProviderByAddress(s.providers, quote.LPRSKAddr)
-	signB, err := p.SignHash(hashBytes)
+	cfuCost, err := s.rsk.EstimateGas(quote.ContractAddr, quote.Value, []byte(quote.Data))
```
Comment (javicid) [inline http/server.go:219]: This should be (quote.gasLimit + CFUExtraGas) * gasPrice + q.value

File: http/watcher.go [hunk 2]
```diff
@@ -140,6 +130,17 @@ func (w *BTCAddressWatcher) performRegisterPegIn(txHash string) error {
...
+		return err
+	}
+	hb, err := hex.DecodeString(h)
+	if err != nil {
+		return err
+	}
+	err = w.lp.RefundLiquidity(hb)
+	if err != nil {
+		return fmt.Errorf("failed to refund to liquidity provider. quote: %v. error: %v", h, err)
+	}
```
Comment (javicid) [inline http/watcher.go:144]: I would put this outside this function. In the same place where close(w.done) is called.

**Review by javicid — CHANGES_REQUESTED** [review]

File: http/watcher.go (hunk 2 above)
Comment (aguschirico) [reply http/watcher.go:152]: done.

File: http/server.go (hunk 1 above)
Comment (aguschirico) [reply http/server.go]: fixed.

**Review by javicid — APPROVED** [review]

**Pull request merged**